scripts/zones/Oldton_Movalpolos/mobs/Goblin_Wolfman.lua

```diff
@@ -11,18 +11,22 @@ local entity = {}
 entity.onMobInitialize = function(mob)
     mob:setMobMod(xi.mobMod.IDLE_DESPAWN, 180)
     mob:addMod(xi.mod.ACC, 70) -- Very accurate
-    mob:setLocalVar("weaponOn", 0)
+    mob:setLocalVar("removeMods", 0)
 end
 
-entity.onMobFight = function(mob, target)
-    local weaponOn = mob:getLocalVar("weaponOn")
-    if mob:hasStatusEffect(xi.effect.BLOOD_WEAPON) and weaponOn == 0 then
+entity.onMobWeaponSkill = function(target, mob, skill)
+    if skill:getID() == xi.jsa.BLOOD_WEAPON then
         mob:addMod(xi.mod.DELAY, 1500)
         mob:addMod(xi.mod.ATTP, 160)
-        mob:setLocalVar("weaponOn", 1)
-    elseif not mob:hasStatusEffect(xi.effect.BLOOD_WEAPON) and weaponOn == 1 then
+        mob:setLocalVar("removeMods", 1)
+    end
+end
+
+entity.onMobFight = function(mob, target)
+    if not mob:hasStatusEffect(xi.effect.BLOOD_WEAPON) and mob:getLocalVar("removeMods") == 1 then
```

Comment:
This can all be condensed into this if statement onMobFight. if mob has blood_weapon, set stats to whatever, else set them to 0.

Reply:
Isn't setting to zero a problem because what if other effects are also moding the same thing? Have to delete right?

Reply:
Yeah perhaps you're right. Only other nitpick I have is that we don't need to set the variable to 0 on initialize. all variables are reverted to 0 when the mob dies.

Reply:
Yep, fixed that, thanks!